core/types/transaction_signing.go

```diff
@@ -159,6 +162,9 @@ func NewSigner(chainId *big.Int) Signer {
 }
 
 func (s SignerV1) Sender(tx *Transaction) (common.Address, error) {
+	if tx.Type() == ExternalTxType { // External TX does not have a signature
+		return common.ZERO, nil
```

Comment:
Actually the ETX struct has a sender field: https://github.com/dominant-strategies/go-quai/blob/b2a635fd11103021146f064940454f90e21cb99d/core/types/external_tx.go#L18

Its ok to use the zero address as a temporary ephemeral sender account for TX processing, but contracts and anyone needing to know the sender of an ETX (i.e. anyone calling this method) should be given the `externalTx.Sender` value.

Reply:
Ok, fixed. Have to write some tests to make sure no functions try to do lookups on the address returned.